internal/context/init.go

```diff
@@ -37,29 +39,45 @@ func Init(outDir string) *projectconfig.ZeroProjectConfig {
 		exit.Fatal("Error creating root: %v ", err)
 	}
 
-	prompts := getProjectPrompts(projectConfig.Name)
-	projectConfig.Parameters["ShouldPushRepoUpstream"] = prompts["ShouldPushRepoUpstream"].GetParam(projectConfig.Parameters)
-	// Prompting for push-up stream, then conditionally prompting for github
-	projectConfig.Parameters["GithubRootOrg"] = prompts["GithubRootOrg"].GetParam(projectConfig.Parameters)
-	personalToken := prompts["githubPersonalToken"].GetParam(projectConfig.Parameters)
-	if personalToken != "" && personalToken != globalconfig.GetUserCredentials(projectConfig.Name).AccessToken {
-		projectConfig.Parameters["githubPersonalToken"] = personalToken
-		projectCredential := globalconfig.GetUserCredentials(projectConfig.Name)
-		projectCredential.GithubResourceConfig.AccessToken = personalToken
-		globalconfig.Save(projectCredential)
-	}
 	moduleSources := chooseStack(getRegistry())
 	moduleConfigs := loadAllModules(moduleSources)
 	for _ = range moduleConfigs {
 		// TODO: initialize module structs inside project
 	}
 
+	prompts := getProjectPrompts(projectConfig.Name, moduleConfigs)
+
+	initParams := make(map[string]string)
+	projectConfig.ShouldPushRepositories = true
+	initParams["ShouldPushRepositories"] = prompts["ShouldPushRepositories"].GetParam(initParams)
+	if initParams["ShouldPushRepositories"] == "n" {
+		projectConfig.ShouldPushRepositories = false
+	}
+
+	// Prompting for push-up stream, then conditionally prompting for github
+	initParams["GithubRootOrg"] = prompts["GithubRootOrg"].GetParam(initParams)
+	initParams["GithubPersonalToken"] = prompts["GithubPersonalToken"].GetParam(initParams)
+	if initParams["GithubRootOrg"] != "" && initParams["GithubPersonalToken"] != globalconfig.GetUserCredentials(projectConfig.Name).AccessToken {
+		projectCredential := globalconfig.GetUserCredentials(projectConfig.Name)
+		projectCredential.GithubResourceConfig.AccessToken = initParams["GithubPersonalToken"]
+		globalconfig.Save(projectCredential)
+	}
+
 	projectParameters := promptAllModules(moduleConfigs)
 	for k, v := range projectParameters {
 		projectConfig.Parameters[k] = v
 		// TODO: Add parameters to module structs inside project
 	}
 
+	for moduleName, _ := range moduleConfigs {
+		// @TODO : Uncomment when this struct is implemented
+		repoName := prompts[moduleName].GetParam(initParams)
+		repoURL := fmt.Sprintf("%s/%s", initParams["GithubRootOrg"], repoName)
+		//projectConfig.Modules[moduleName].Files.Directory = prompts[moduleName].GetParam(initParams)
+		//projectConfig.Modules[moduleName].Files.Repository = repoURL
+		fmt.Println(repoURL)
```

Comment:
stray print command

Reply:
Yeah, reminder that that code needs to be filled in.